src/Workspace/index.jsx

```diff
@@ -328,6 +329,7 @@ class Workspace extends React.Component {
         path: `${workspaceTerminateUrl}`,
         method: 'POST',
       }).then(() => {
+        this.state.initiatePayModelCallUponTermination = true;
```

Comment:
minor: i don't think this has to be a state variable, should be able to make it as an argument that passes into `checkWorkspaceStatus()`
this can save some re-render cycles caused by unnecessary state updates

Reply:
I think this makes sense. We can avoid unnecessary state updates if we make it as an argument to `checkWorkspaceStatus()`. I will work on that.

Reply:
I've made the proposed changes and pushed a new commit. Please review and let me know if everything looks good. @mfshao